### PR TITLE
openssh: update to 9.3p1

### PR DESCRIPTION
**Pull request by artkiver**
###### Type(s)

- [x] bugfix
- [x] enhancement
- [x] security fix

###### Tested on

macOS 13.2.1 22D68 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by artkiver** [comment]
Was running into an error with the launchd.patch, whereas omitting it seemed to build cleanly. It seems as if it can be omitted safely (and perhaps the others too as the HPN variant has been removed?) but additional refactoring may be warranted. For the time being given that there were some security fixes, I thought better to offer a PR that seems to work for now and tidy up more later as required.

File: net/openssh/Portfile [hunk 1]
```diff
@@ -50,8 +50,7 @@ if {${name} eq ${subport}} {
 
     # the HPN patch needs this, so rewrite all other patches to support it, too
     patch.args          -p1
-    patchfiles          launchd.patch \
```
Comment (ryandesign) [inline net/openssh/Portfile:53]: Should this patchfile also be deleted?

**Review by ryandesign — COMMENTED** [review]
The period at the end of the commit message could be removed.

**Comment by ryandesign** [comment]
> Was running into an error with the launchd.patch

What error?

**Comment by ryandesign** [comment]
> It seems as if it can be omitted safely (and perhaps the others too as the HPN variant has been removed?

If you are referring to this comment:

https://github.com/macports/macports-ports/blob/a61fcbaaaa5e21fedb2997fda946c149bd00e915/net/openssh/Portfile#L51-L52

then I believe the way to interpret the comment is that the hpn patch used p1, therefore all patches had to use p1. Now that the hpn patch is no longer being used, the patches could be edited to use p0 instead, if you desire. The comment should not be interpreted as meaning that all of the patchfiles were only needed for the hpn variant.

**Comment by artkiver** [comment]
> > Was running into an error with the launchd.patch
> 
> What error?

Thanks for the extra set of eyeballs!

With minor modifications to the Portfile (without omitting the launchd.patch as in the submitted PR), port -v install is as follows:

```
 port -v install       
--->  Computing dependencies for openssh.
--->  Fetching archive for openssh
--->  openssh-9.3p1_0.darwin_22.arm64.tbz2 doesn't seem to exist in /opt/local/var/macports/incoming/verified
--->  Attempting to fetch openssh-9.3p1_0.darwin_22.arm64.tbz2 from http://mirror.fcix.net/macports/packages/openssh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0   153    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch openssh-9.3p1_0.darwin_22.arm64.tbz2 from https://packages.macports.org/openssh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0   126    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Attempting to fetch openssh-9.3p1_0.darwin_22.arm64.tbz2 from https://ywg.ca.packages.macports.org/mirror/macports/packages/openssh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0   153    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
--->  Fetching distfiles for openssh
--->  Verifying checksums for openssh
--->  Checksumming openssh-9.3p1.tar.gz
--->  Extracting openssh
--->  Extracting openssh-9.3p1.tar.gz
Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_openssh/openssh/work" && /usr/bin/gzip -dc '/opt/local/var/macports/distfiles/openssh/openssh-9.3p1.tar.gz' | /usr/bin/tar -xf - 
--->  Applying patches to openssh
--->  Applying launchd.patch
Executing:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_openssh/openssh/work/openssh-9.3p1" && /usr/bin/patch -p0 -p1 < '/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/net/openssh/files/launchd.patch'
patching file clientloop.c
patching file channels.c
patching file channels.h
1 out of 1 hunks failed--saving rejects to channels.h.rej
Command failed:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_openssh/openssh/work/openssh-9.3p1" && /usr/bin/patch -p0 -p1 < '/opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/net/openssh/files/launchd.patch'
Exit code: 1
Error: Failed to patch openssh: command execution failed
Error: See /opt/local/var/macports/logs/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_openssh/openssh/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there
is a bug.
Error: Processing of port openssh failed
```
[main.log](https://github.com/macports/macports-ports/files/10987313/main.log)

I thought to open a Trac ticket to attempt to attach the channels.h.rej as referenced above since GitHub threw an error, but it seems as if Trac is also giving timeouts, so I copy and pasted the text there:

https://trac.macports.org/ticket/67109

**Comment by artkiver** [comment]
[Portfile.9.3p1.causeserrors.txt](https://github.com/macports/macports-ports/files/10987412/Portfile.9.3p1.causeserrors.txt)

File: net/openssh/Portfile (hunk 1 above)
Comment (artkiver) [reply net/openssh/Portfile:53]: > Should this patchfile also be deleted?

That would probably be sensible as well, yes. Possibly some of the other patch files as well if they were related to HPN?

**Comment by artkiver** [comment]
> > It seems as if it can be omitted safely (and perhaps the others too as the HPN variant has been removed?
> 
> If you are referring to this comment:
> 
> https://github.com/macports/macports-ports/blob/a61fcbaaaa5e21fedb2997fda946c149bd00e915/net/openssh/Portfile#L51-L52
> 
> then I believe the way to interpret the comment is that the hpn patch used p1, therefore all patches had to use p1. Now that the hpn patch is no longer being used, the patches could be edited to use p0 instead, if you desire. The comment should not be interpreted as meaning that all of the patchfiles were only needed for the hpn variant.

That seems cogent with my interpretation of that comment as well, though I didn't want to unilaterally go about dispensing with everything without a bit more testing and collaboration, so thank you for your perspective too! My thinking was to provide something functional with a bit of expediency and then with additional effort and revisions clean things up further, but I will defer to your expertise on such matters.

Thanks!

**Comment by ryandesign** [comment]
> ```
> 1 out of 1 hunks failed--saving rejects to channels.h.rej
> ```

Ok, so the patch just needs to be updated:

* Open the patch and see what it's doing (adding some lines between some other lines)
* Open the file being patched and try to find the relevant spot in the code
* Add in the new lines
* Generate a new patchfile

I've added a commit to this PR to do that. When we merge the PR, we should use the "Squash and merge" button to combine the two commits into one.

**Review by artkiver — COMMENTED** [review]
Tested and looks good! Thank you @ryandesign for the assistance and improvement!🙏

**Pull request merged**